src/state/ProcessingState.cpp

```diff
@@ -211,13 +203,24 @@ int ProcessingState::runStateProcess()
     // IO writing
     unsigned char *rgbOut = convertMaskToRGB(output, width * height, width, height);
     // unsigned char* pixelOut = converToPixelMask(rgbOut, width, height, 3);
-    
-    ExportImage oMask(rgbOut, width, height, 3, "output.jpg",this->logger);
+
+    // arbitrary decision logic
+    const int DECISION_THRESHOLD = 80;
```

Comment:
Looks good, however the filter classes have a decision maker function for them (not implemented) maybe it would be better to move it there? otherwise looks great